### PR TITLE
Round widget result values

### PR DESCRIPTION
**Pull request by laem**
Fixes issue #84

### CODE REVIEW

File: js/ui.js [hunk 1]
```diff
@@ -13,8 +13,14 @@ function display(data) {
 		if (! target)
 			return
 
-		if (typeof value == 'number')
-			value = String(value.toFixed(2)).replace('.', ',')
+		if (typeof value == 'number'){
```
Comment (MattiSG) [inline js/ui.js]: Space before curly brace.

> Add [`space-before-blocks: 2`](http://eslint.org/docs/rules/space-before-blocks.html) to ESlint.

File: js/ui.js [hunk 2]
```diff
@@ -13,8 +13,14 @@ function display(data) {
...
+			if (target.hasAttribute('data-round')){
+				value = Math.round(value)
+			} else {
```
Comment (MattiSG) [inline js/ui.js]: Inconsistent use of curly braces for `if`-`else` constructs.

> Add [`curly: [2, "multi-or-nest"]`](http://eslint.org/docs/rules/curly.html) to ESlint.

File: js/ui.js [hunk 3]
```diff
@@ -13,8 +13,14 @@ function display(data) {
 		if (! target)
 			return
 
-		if (typeof value == 'number')
-			value = String(value.toFixed(2)).replace('.', ',')
+		if (typeof value == 'number'){
+			if (target.hasAttribute('data-round')){
+				value = Math.round(value)
```
Comment (MattiSG) [inline js/ui.js]: Wouldn't this be simpler with a ternary: `value = value.toFixed(target.hasAttribute('data-round') ? 0 : 2)`.

File: js/ui.js [hunk 4]
```diff
@@ -13,8 +13,14 @@ function display(data) {
 		if (! target)
 			return
 
-		if (typeof value == 'number')
-			value = String(value.toFixed(2)).replace('.', ',')
+		if (typeof value == 'number') {
+			if (target.hasAttribute('data-round'))
+				value = Math.round(value)
+			else
+				value = value.toFixed(2)
```
Comment (MattiSG) [inline js/ui.js]: Wouldn't this be simpler with a ternary: `value = value.toFixed(target.hasAttribute('data-round') ? 0 : 2)`.

Comment (MattiSG) [inline js/ui.js]: Or `value = value.toFixed(target.attributes['data-decimals'] || 2)`, with `data-decimals="0"` in HTML.

**Comment by laem** [comment]
> Or value = value.toFixed(target.attributes['data-decimals'] || 2), with data-decimals="0" in HTML.

I prefer your ternary, since data-decimals = 0 does't mean it's going to be rounded.
`toFixed` returns a string, so no need to `String()` in the next line.

**Comment by MattiSG** [comment]
> toFixed returns a string, so no need to String() in the next line.
Oh, nice one!  :)

File: js/ui.js [hunk 5]
```diff
@@ -13,8 +13,11 @@ function display(data) {
 		if (! target)
 			return
 
-		if (typeof value == 'number')
-			value = String(value.toFixed(2)).replace('.', ',')
+		if (typeof value == 'number') {
+			value = value
+				.toFixed(target.hasAttribute('data-round') ? 0 : 2)
+				.replace('.', ',')
```
Comment (MattiSG) [inline js/ui.js]: Really neat!  :+1:

**Comment by MattiSG** [comment]
GTM.

I'll rerun the build manually, SauceLabs had locked us out, hence the CI fail. I've fixed it now.

**Pull request merged**